charmhelpers/core/hookenv.py

```diff
@@ -1088,22 +1097,31 @@ def function_tag():
     return os.environ.get('JUJU_FUNCTION_TAG') or action_tag()
 
 
-def status_set(workload_state, message):
+def status_set(workload_state, message, application_status=False):
```

Comment:
IMO there's no need for this parameter to have the `_status` suffix.

Reply:
I'll address this too